prov/usnic/src/usdf_pep.c

```diff
@@ -186,37 +186,39 @@ usdf_pep_read_connreq(void *v)
 	/* if resid is 0 now, completely done */
 	if (crp->cr_resid == 0) {
 		ret = usdf_pep_creq_epoll_del(crp);
-		if (ret != 0) {
-			usdf_cm_msg_connreq_failed(crp, ret);
-			return 0;
-		}
+		if (ret != 0)
+			goto report_failure_skip_data;
 
 		/* create CONNREQ EQ entry */
 		entry_len = sizeof(*entry) + reqp->creq_datalen;
 		entry = malloc(entry_len);
 		if (entry == NULL) {
-			usdf_cm_msg_connreq_failed(crp, -errno);
-			return 0;
+			ret = -errno;
+			goto report_failure_skip_data;
 		}
 
 		entry->fid = &pep->pep_fid.fid;
 		entry->info = usdf_pep_conn_info(crp);
-		if (entry->info == NULL) {
-			free(entry);
-			usdf_cm_msg_connreq_failed(crp, -FI_ENOMEM);
-			return 0;
-		}
+		if (entry->info == NULL)
+			goto free_entry_and_report_failure;
```

Comment:
> What's the value of ret here?

Hmm, missed that. Thanks. `usdf_pep_conn_info` seems to be only be able to fail in the event of a failure in `calloc`, so I'll modify that to `-FI_ENOMEM`.